DOC added reference to plot_grid_search_stats.py

Cheath2 opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->

Towards #30621 

#### What does this implement/fix? Explain your changes.
Added a reference to plot_grid_search_stats.py within the User manuals 'Analyzing results with the `cv_results_` attribute' section.

#### Any other comments?


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
https://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

Cheath2 committed:
added reference

StefanieSenger reviewed (changes requested):
Hello @Cheath2,

thanks for making this PR. 

To be honest, I believe this link would be misplaced in the Halving*Search section. This section and the example both deal with `cv_results_`, but the section where you linked it focuses on explaining the structure of the results in Halving, while the example you linked is more about visualizing and interpreting GridSearchCV results graphically. 
Also, the example is already linked above in the GridSearch section (line 106), where it seems better placed.

Since the example shows how to plot and interpret results using GridSearch's `cv_results_` attribute, I would rather suggest to add it to the docstring of GridSearchCV in the attribute description of `cv_results_`.
Also probably change the description from "in-depth statistical analysis" to "visualization and interpretation of GridSearch results" or something similar.

Would you be up to do that? If so, you would also need to remove the `desktop.ini` file from your PR and only push the actual change. Thank you.

Cheath2 committed:
updating  per instruction

Cheath2 committed:
Merge branch 'scikit-learn:main' into plotgridsearchstats

Cheath2 committed:
Merge branch 'plotgridsearchstats' of https://github.com/Cheath2/scikit-learn into plotgridsearchstats

Cheath2 committed:
removed bad edit

Cheath2 commented:
Hello @StefanieSenger ,

Thank you for your guidance, I believe I have updated the PR per your instruction. I updated the verbiage, moved the example reference to the attribute description within cv_results_ docstring under the existing reference text, and addressed the .ini issue.  Please let me know if any other changes are needed.

StefanieSenger reviewed (approved):
Thank you @Cheath2, that looks very good.

Can we merge this, @adrinjalali?

StefanieSenger commented:
Can you please add a DOC in from of your PR title, @Cheath2?

adrinjalali commented on sklearn/model_selection/_search.py:
```diff
@@ -1446,6 +1446,9 @@ class GridSearchCV(BaseSearchCV):
             'params'             : [{'kernel': 'poly', 'degree': 2}, ...],
             }
 
+        For an example of visualization and interpretation of GridSearch results,
+        see :ref:`sphx_glr_auto_examples_model_selection_plot_grid_search_stats.py`.
```
I like where this link is, but it could be also added to all the other 3 search cv classes where `cv_results_` are present.

marenwestermann committed:
Merge remote-tracking branch 'upstream/main' into plotgridsearchstats

marenwestermann committed:
address comments

marenwestermann committed:
add back ticks and shorten description

marenwestermann reviewed (commented):
I updated the PR and addressed the comments. Thank you @Cheath2 for your work on this PR!
